### PR TITLE
Update readme and release notes

### PR DESCRIPTION
**Pull request by MaximProshin**
### Changes

- Minor update in Readme to have links on one line
- Remove numpy2.0 from release notes to avoid any confusion

### CODE REVIEW

**Review by KodiaqQ — APPROVED** [review]

**Comment by KodiaqQ** [comment]
@MaximProshin, shouldn't we update `docs/PyPiPublishing.md` also?

**Comment by MaximProshin** [comment]
> @MaximProshin, shouldn't we update `docs/PyPiPublishing.md` also?

There is no such links in the header there or did I miss anything?

File: README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@
 [Usage](#usage) •
 [Tutorials and Samples](#demos-tutorials-and-samples) •
 [Third-party integration](#third-party-repository-integration) •
-[NNCF Model Zoo](./docs/ModelZoo.md)
+[Model Zoo](./docs/ModelZoo.md)
```
Comment (KodiaqQ) [inline README.md:11]: Shouldn't we align https://github.com/openvinotoolkit/nncf/blob/develop/docs/PyPiPublishing.md?plain=1#L23 also?

Comment (MaximProshin) [reply README.md:11]: No, the change was made to just avoid multiple lines in the header while NNCF Model Zoo is much more clear than Model Zoo, which we intended to change before.

**Pull request merged**